### PR TITLE
fix(gcrouter): breaking changes removed * from access control allow o…

### PR DESCRIPTION
**Pull request by taran1898**

### CODE REVIEW

File: lib/router.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const gc_logger_1 = require("@adhityan/gc-logger");
 const helmet_1 = __importDefault(require("helmet"));
 const middleware_1 = require("./middleware");
 class Router {
-    static initialize(app, config, container) {
+    static initialize(app, config, container, allowedOrigin) {
```
Comment (suresh-saini) [inline lib/router.js:15]: please do the changes to ts files. js files are created by compiling ts files

File: lib/middleware.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ let ErrorFormatHandler = class ErrorFormatHandler {
             delete error.errorIdentifier;
         if (error.stack)
             delete error.stack;
-        response.status(status).send(Object.keys(error).length === 0 ? { message: error.message } : error);
+        response.status(status).json(Object.keys(error).length === 0 ? { message: error.message } : error);
```
Comment (suresh-saini) [inline lib/middleware.js:20]: please do the changes to ts files. js files are created by compiling ts files

File: src/router.ts [hunk 3]
```diff
@@ -46,7 +46,8 @@ export class Router {
 
                 res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
                 res.setHeader('Access-Control-Allow-Methods', 'GET');
-                res.setHeader('Access-Control-Allow-Origin', '*');
+                if (allowedOrigin)
+                    res.setHeader('Access-Control-Allow-Origin', `${allowedOrigin}`);
```
Comment (suresh-saini) [inline src/router.ts:50]: why don't you use `allowedOrigin` directly?

**Review by suresh-saini — CHANGES_REQUESTED** [review]
please also run build so that generated files are updated based on ts changes

File: src/router.ts (hunk 3 above)
Comment (taran1898) [reply src/router.ts:50]: We can't have null as a value for this. Refer doc https://w3c.github.io/webappsec-cors-for-developers/#avoid-returning-access-control-allow-origin-null

**Comment by taran1898** [comment]
> please also run build so that generated files are updated based on ts changes

Already done.

File: lib/middleware.js (hunk 2 above)
Comment (suresh-saini) [reply lib/middleware.js:20]: I am not saying about condition. I am saying about ``${allowedOrigin}\``

File: src/router.ts (hunk 3 above)
Comment (suresh-saini) [reply src/router.ts:50]: I am not saying about condition. I am saying about ``${allowedOrigin}\``

**Review by suresh-saini — APPROVED** [review]

**Pull request merged**